SqrMelon/animationgraph/curvedata.py

```diff
@@ -321,21 +321,30 @@ def _evaluate(lhs_time: float, lhs_value: float, lhs_tangent: float, rhs_tangent
         t = (time - lhs_time) / dx
         return t * (t * (t * c0 + c1) + c2) + c3
 
+    def evaluateWithSnapAndKey(self, time: float, precision: float) -> List[Union[float, int]]:
```

Comment:
This is always 1 float and 2 ints, so the return type can be tuple[float, int, int] to be more exact.